src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -76,113 +76,104 @@ const unsigned mode_flag_custom = 1;
 
 using namespace time_literals;
 
-mavlink_hil_actuator_controls_t Simulator::actuator_controls_from_outputs()
+void Simulator::actuator_controls_from_outputs(mavlink_hil_actuator_controls_t *msg)
 {
-	mavlink_hil_actuator_controls_t msg{};
+	memset(msg, 0, sizeof(mavlink_hil_actuator_controls_t));
 
-	msg.time_usec = hrt_absolute_time() + hrt_absolute_time_offset();
+	msg->time_usec = hrt_absolute_time() + hrt_absolute_time_offset();
 
 	bool armed = (_vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED);
 
-	const float pwm_center = (PWM_DEFAULT_MAX + PWM_DEFAULT_MIN) / 2;
-
 	int _system_type = _param_mav_type.get();
 
-	/* scale outputs depending on system type */
-	if (_system_type == MAV_TYPE_AIRSHIP ||
-	    _system_type == MAV_TYPE_QUADROTOR ||
-	    _system_type == MAV_TYPE_HEXAROTOR ||
-	    _system_type == MAV_TYPE_OCTOROTOR ||
-	    _system_type == MAV_TYPE_VTOL_DUOROTOR ||
-	    _system_type == MAV_TYPE_VTOL_QUADROTOR ||
-	    _system_type == MAV_TYPE_VTOL_TILTROTOR ||
-	    _system_type == MAV_TYPE_VTOL_RESERVED2) {
-
-		/* multirotors: set number of rotor outputs depending on type */
+	unsigned motors_count;
+	bool is_fixed_wing;
 
-		unsigned n;
-
-		switch (_system_type) {
-		case MAV_TYPE_AIRSHIP:
-		case MAV_TYPE_VTOL_DUOROTOR:
-			n = 2;
-			break;
+	switch (_system_type) {
+	case MAV_TYPE_AIRSHIP:
+	case MAV_TYPE_VTOL_DUOROTOR:
+	case MAV_TYPE_COAXIAL:
+		motors_count = 2;
+		is_fixed_wing = false;
```

Comment:
Can you just initialize this to `false`? Then you don't have to set it in most cases.

Reply:
I think its better as I write it. it ensures setting is_fixed_wing to true or false.
there is no issue of memory or performance here, its more issue of clearly

Reply:
@bkueng , you don't agree?

Reply:
Personally I would reduce the amount of code, but not super important here.